app/src/org/commcare/views/media/ExpandedAudioPlaybackView.java

```diff
@@ -0,0 +1,149 @@
+package org.commcare.views.media;
+
+import android.animation.ObjectAnimator;
+import android.content.Context;
+import android.os.Build;
+import android.os.Handler;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.animation.LinearInterpolator;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import org.commcare.dalvik.R;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Audio playback widget with clickable horizontal progress bar
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class ExpandedAudioPlaybackView extends AudioPlaybackButtonBase {
+    private ProgressBar seekBar;
+    private ObjectAnimator animation;
+    private Handler handler;
+    private TextView progressText;
+    private int playbackDurationMillis;
+
+    public ExpandedAudioPlaybackView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+
+        progressText = (TextView)findViewById(R.id.duration_info);
+    }
+
+    public ExpandedAudioPlaybackView(Context context, String URI) {
+        super(context, URI, null, true);
+
+        progressText = (TextView)findViewById(R.id.duration_info);
+    }
+
+    @Override
+    protected int getLayout() {
+        return R.layout.expanded_audio_playback;
+    }
+
+    @Override
+    protected void startProgressBar(int currentPositionMillis, int milliDuration) {
+        playbackDurationMillis = milliDuration;
+        setupProgressBar();
+        setupProgressAnimation(currentPositionMillis);
+        launchElapseTextUpdaterThread();
+    }
+
+    private void setupProgressBar() {
+        seekBar = (ProgressBar)findViewById(R.id.seek_bar);
+        seekBar.setEnabled(true);
+        seekBar.setMax(playbackDurationMillis);
+        seekBar.setOnTouchListener(
+                new OnTouchListener() {
+                    @Override
+                    public boolean onTouch(View v, MotionEvent event) {
+                        return performProgressBarTouch(v, event);
+                    }
+                });
+    }
+
+    private boolean performProgressBarTouch(View v, MotionEvent event) {
+        int progress = (int)(playbackDurationMillis * (event.getX() / v.getWidth()));
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
+            animation.setCurrentPlayTime(progress);
+        }
+        updateProgressText(progress, playbackDurationMillis);
+        if (AudioController.INSTANCE.mediaForInstance(ExpandedAudioPlaybackView.this)) {
+            AudioController.INSTANCE.seekTo(progress);
+        }
+        return false;
+    }
+
+    private void setupProgressAnimation(int currentPositionMillis) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
+            final int startPosition = 0;
+            animation = ObjectAnimator.ofInt(seekBar, "progress", startPosition, playbackDurationMillis);
```

Comment:
So I think the way you did things here is what I was originally expecting in `startProgressBar` of AudioPlaybackButton, and what I was trying but failing to express well in our discussion on that PR. In that one, you picked an arbitrary value for the `progressDuration` value that is passed to `ofInt` (https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/views/media/AudioPlaybackButton.java#L303), whereas here you use the duration of the audio that is being played. Looking back at your explanation on that PR, it seems like what you did there might actually be more correct, in that you want to make sure that `progressDuration` is a large value in order to create a smooth animation, and it doesn't really matter if that value corresponds to the audio playback's length. Do you still think that's right?